bin/complete.rb

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env ruby
+
+# Thor / ActiveSupport path magic
+rmt_path = File.expand_path('..', __dir__)
+require_relative '../config/boot'
+$LOAD_PATH.unshift File.join(rmt_path, 'lib')
+
+require 'active_support'
+
+relative_load_paths = %w[lib lib/rmt/cli/].map { |dir| File.join(rmt_path, dir) }
+ActiveSupport::Dependencies.autoload_paths += relative_load_paths
+# magic ends here
+
+# helper functions:
+def get_options(command)
+  submodule = command.slice(0,1).capitalize + command.slice(1, command.length).downcase
+  begin
+    options = RMT::CLI.module_eval(submodule).commands.keys
+  rescue NameError
+    options = []
+  end
+end
+
+def split_feed
+  words = ENV['COMP_LINE'].split(' ')
+  index = words.length - 1
+  if ENV['COMP_LINE'][-1] == ' '
```

Comment:
In which case would you even a space in the array if you are splitting by space? :thinking:

Reply:
Ah, it's `ENV['COMP_LINE']`. But I'm still not sure what this does.

Reply:
COMP_LINE are all characters currently typed in the CLI feed. When I split by spaces, I lose trailing spaces. But those indicate that the previous word is done and the user is looking for the completion of the next sub-command, which means, the current word is an empty string.

This is handled by the assignemnt:
```ruby
current_word = words[index] || ''
```